tests/lib/controllers/tokenize-controller.test.js

```diff
@@ -89,4 +90,73 @@ describe('tokenize-controller.test.js', () => {
     expect(resultOptions.tei).toEqual(expect.any(Options))
   })
 
+  it('7 TokenizeController - getNextTokenIdWordChangesType - check idWords by workflow for the same token with different changes', async () => {
+    // update token's word - the first time
+    const newWordId1 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: '1-0-0',
+      changeType: TokensEditController.changeType.UPDATE
+    })
+    expect(newWordId1).toEqual('1-0-0-e-1')
+
+    // update token's word - the second time
+    const newWordId2 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId1,
+      changeType: TokensEditController.changeType.UPDATE
+    })
+    expect(newWordId2).toEqual('1-0-0-e-2')
+
+    // merge this token with another one
+    const newWordId3 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId2,
+      changeType: TokensEditController.changeType.MERGE
+    })
+    expect(newWordId3).toEqual('1-0-0-e-2-m-1')
+
+    // merge this token with another one the second time
+    const newWordId4 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId3,
+      changeType: TokensEditController.changeType.MERGE
+    })
+    expect(newWordId4).toEqual('1-0-0-e-2-m-2')
+
+    // split this token to two token - this is the idWord for the first part
+    const newWordId5 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId4,
+      changeType: TokensEditController.changeType.SPLIT,
+      indexWord: 1
+    })
+
+    expect(newWordId5).toEqual('1-0-0-e-2-m-2-s1-1')
+
+    // split this token to two token again - this is the idWord for the first part again
+    const newWordId6 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId5,
+      changeType: TokensEditController.changeType.SPLIT,
+      indexWord: 1
+    })
+    expect(newWordId6).toEqual('1-0-0-e-2-m-2-s1-2')
+
+    // split this token to two token again - this is the idWord for the second part
+    const newWordId7 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId6,
+      changeType: TokensEditController.changeType.SPLIT,
+      indexWord: 2
+    })
+    expect(newWordId7).toEqual('1-0-0-e-2-m-2-s1-2-s2-1')
+
+    // merge this token with another one again
+    const newWordId8 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId7,
+      changeType: TokensEditController.changeType.MERGE
+    })
+    expect(newWordId8).toEqual('1-0-0-e-2-m-2-s1-2-s2-1-m-1')
+
+    // update this token with another one again
+    const newWordId9 = TokenizeController.getNextTokenIdWordChangesType({
+      tokenIdWord: newWordId8,
+      changeType: TokensEditController.changeType.UPDATE
+    })
+    expect(newWordId9).toEqual('1-0-0-e-2-m-2-s1-2-s2-1-m-1-e-1')
+
+  })

```

Comment:
This is the tests case for new idWord workflow

Reply:
I know it looks insane :-) maybe it is.  I'm not sure if it will be useful or not. I hope so.